### PR TITLE
Accounts limit access

### PR DESCRIPTION
**Pull request by firyx**
# PULL REQUEST

## Overview

Update `.env` generation with `ACCOUNTS_LIMIT_ACCESS`.

## Checklist
Review and complete the checklist to ensure that the PR is complete before assigned to an approver.
 - [x] All new methods or updated methods have clear docstrings
 - [x] Testing added or updated for new methods
 - [x] Verify if any changes impact the WebPortal Health Checks
 - [x] Approriate documentation updated
 - [x] Changelog file created

### CODE REVIEW

File: playbooks/templates/.env.j2 [hunk 1]
```diff
@@ -131,6 +131,14 @@ SKYNET_DASHBOARD_URL=https://account.{{ cluster_domain }}
 # Accounts log level
 SKYNET_ACCOUNTS_LOG_LEVEL=info
 
+{% if webportal_common_config.accounts_limit_access is defined %}
+# Accounts limit access to authenticated users
+# Possible values:
+# - 'authenticated': Only authenticated users can access the portal (default if this env variable is missing)
+# - 'false' or empty: All users incl. anonymous users can access the portal (access to portal is public)
```
Comment (ro-tex) [inline playbooks/templates/.env.j2:138]: We have this in docker-compose:
```
- ACCOUNTS_LIMIT_ACCESS=${ACCOUNTS_LIMIT_ACCESS:-authenticated} #
```
which I think means that the empty value will be replaced with `authenticated`. I'm not 100% sure about that.

**Review by ro-tex — APPROVED** [review]

File: playbooks/templates/.env.j2 (hunk 1 above)
Comment (firyx) [reply playbooks/templates/.env.j2:138]: Oh, you are right, I have read a comment on the variable, not the code.

File: playbooks/templates/.env.j2 [hunk 2]
```diff
@@ -131,6 +131,14 @@ SKYNET_DASHBOARD_URL=https://account.{{ cluster_domain }}
 # Accounts log level
 SKYNET_ACCOUNTS_LOG_LEVEL=info
 
+{% if webportal_common_config.accounts_limit_access is defined %}
+# Accounts limit access to authenticated users
+# Possible values:
+# - 'authenticated': Only authenticated users can access the portal (default if unset or empty)
+# - 'false': All users incl. anonymous users can access the portal (access to portal is public)
+ACCOUNTS_LIMIT_ACCESS={{ webportal_common_config.accounts_limit_access }}
```
Comment (MSevey) [inline playbooks/templates/.env.j2:139]: if the intention is for `authenticated` to be the default, shouldn't we just set it like this:
```
ACCOUNTS_LIMIT_ACCESS={{ webportal_common_config.accounts_limit_access | default('authenticated') }}
```
and leave out the if?

Or is the default behavior of ACCOUNTS_LIMIT_ACCESS being defined somewhere else?
If it is being defined somewhere else, I would extend the comment to say that X system handle the default behavior setting so ansible should not handle the default case, and on needs to respond to if the flag is set or not.

Comment (firyx) [reply playbooks/templates/.env.j2:141]: Default value if empty or unset is handled by `skynet-webportal/docker-compose.accounts.yml`.
The comment was updated.

**Review by ro-tex — APPROVED** [review]

**Review by MSevey — APPROVED** [review]

**Pull request merged**